snapshot/kaniko_snapshot_unix.go

```diff
@@ -0,0 +1,107 @@
+// +build linux darwin
+
+package snapshot
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/buildpacks/lifecycle"
+
+	kconfig "github.com/GoogleContainerTools/kaniko/pkg/config"
+	ksnap "github.com/GoogleContainerTools/kaniko/pkg/snapshot"
+	kutil "github.com/GoogleContainerTools/kaniko/pkg/util"
+)
+
+type KanikoSnapshotter struct {
+	RootDir                    string
+	DetectFilesystemIgnoreList bool
+	IgnoredPaths               []string
+	snapshotter                *ksnap.Snapshotter
+}
+
+func NewKanikoSnapshotter(rootDir string) (lifecycle.LayerSnapshotter, error) {
+	ls := KanikoSnapshotter{
+		RootDir:                    rootDir,
+		DetectFilesystemIgnoreList: true,
+		IgnoredPaths: []string{
+			"/tmp",
+			"/layers",
```

Comment:
Thinking about this more... does it make sense to hard code `/layers`, since it can change? (If a different value is provided for `-layers` or `CNB_LAYERS_DIR`) 

Same for `/platform`...

Reply:
ah, yeah. Good call. I think we may need to thread through the configurable bits here and make this more dynamic.